staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -382,25 +382,22 @@ func (o *DebugOptions) visitPod(ctx context.Context, pod *corev1.Pod) (*corev1.P
 // debugByEphemeralContainer runs an EphemeralContainer in the target Pod for use as a debug container
 func (o *DebugOptions) debugByEphemeralContainer(ctx context.Context, pod *corev1.Pod) (*corev1.Pod, string, error) {
 	pods := o.podClient.Pods(pod.Namespace)
-	ec, err := pods.GetEphemeralContainers(ctx, pod.Name, metav1.GetOptions{})
+	klog.V(2).Infof("existing ephemeral containers: %v", pod.Spec.EphemeralContainers)
+
+	debugContainer := o.generateDebugContainer(pod)
+	klog.V(2).Infof("new ephemeral container: %#v", debugContainer)
+
+	pod.Spec.EphemeralContainers = append(pod.Spec.EphemeralContainers, *debugContainer)
+	result, err := pods.UpdateEphemeralContainers(ctx, pod.Name, pod, metav1.UpdateOptions{})
```

Comment:
if the pod has changed in the meantime, this will return a conflict error. should we submit a patch to /ephemeralcontainers instead that sets the `.spec.ephemeralContainers` list?

Reply:
I agree that would be better. I switched this to a strategic merge patch.